.github/workflows/test.yml

```diff
@@ -4,18 +4,21 @@ on:
   push:
 
 env:
-  CURRENT_STABLE_CHANNEL: nixpkgs-23.11-darwin
+  CURRENT_STABLE_CHANNEL: nixpkgs-24.05-darwin
 
 jobs:
   test-stable:
     runs-on: macos-12
     timeout-minutes: 30
     steps:
     - uses: actions/checkout@v3
+    # We use the Determinate Systems installer for 2.18 because the
+    # Sequoia UID/GID changes have not yet been backported to the
+    # official installer for that version.
     - name: Install nix corresponding to latest stable channel
-      uses: cachix/install-nix-action@v23
+      uses: DeterminateSystems/nix-installer-action@main
       with:
-        install_url: https://releases.nixos.org/nix/nix-2.13.6/install
+        nix-package-url: https://releases.nixos.org/nix/nix-2.18.5/nix-2.18.5-x86_64-darwin.tar.xz
```

Comment:
I think it would be better to have a job for each installer rather than replacing the installer in this job

Reply:
We have no choice until the Nix team release a 2.18 with the fix, per the comment. This is just a hack taking advantage of the fact that the DetSys installer already deployed the fix and can install multiple Nix versions.

Reply:
Currently we're still running the tests on macOS 12, so adding a second job seems fine to me, even if the Nix installer doesn't support the Sequoia UID migration yet

Reply:
The test will fail because this PR means it will print the message informing that the migration script has to be run and error out. Which is the correct behaviour to ensure that users are prepared for the Sequoia update, but doesn't test what we want.

Note that the upgrade breaks Nix entirely if no migration was done and the user will not even be able to activate nix-darwin to see the message. That's why it's urgent that people are informed and run the migration script *before* upgrading to Sequoia.

Reply:
How about in a different job running the official Nix installer then the migration script in CI and then try to do installation again? If this is too much extra work, we can leave it out

Reply:
That could work, but it seemed more fiddly than just changing a few lines until the patched installer was out. If you'd prefer that I can try to implement it tomorrow, but I'll be busy for most of the day and I'm worried about how many users we'll be able to get this change in front of before Sequoia drops. I lean towards iterating on non-essential things after merge.